### ISSUE TITLE
Enhancement: Semantic search for conversation

### ISSUE THREAD

**Issue by HyunggyuJang** [danny-avila/LibreChat#618]
### What features would you like to see added?

Currently, meilisearch (exact match) is used for search feature. It would be interesting to see supporting semantic search using vector embedding.

### More details

If we can support it, the next stage would be to enable chatbot to access the semantic search result so that all conversations can interact each other.

### Which components are impacted by your request?

General

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Yes, this is my next big push on the project. I'm wrapping up handling ChatGPT plugins/openAPI specs. Just need to write documentation for this. Then going all in on file indexing/vector integrations. 

However, I would like to mention that vector retrieval of documents will be prioritized over messages. Messages, especially contextually will be tricky, unless you explicitly ask the LLM for a look back at previous messages based on a semantic search. I think this last point is the move forward, because then it can function reliably as a plugin. 

Ideally, every endpoint/AI provider could benefit from this and I will have to sit down and think of the best design patterns for this. This is kind of a separate point but a necessary refactor moving forward.

A semantic search bar of all messages, similar to meilisearch, would probably be after all the things above.

I've been actively researching all the necessary tools for this and I'm just finalizing on the big picture

**Comment by danny-avila** [comment]
Triage note on a 3-year-old request. Still valid and still unimplemented: conversation search is full-text via MeiliSearch, and there is no vector search over messages.

What did land is the retrieval side for files (`RAG_API_URL`, `EMBEDDINGS_PROVIDER`, `EMBEDDINGS_MODEL` are all live config), which matches the ordering @danny-avila laid out above: documents first, messages later. So the embedding infrastructure now exists, it just is not wired to conversation history.

A fix would touch the MeiliSearch plugin (`packages/data-schemas/src/models/plugins/mongoMeili.ts`), the search route, and the RAG API for message-level embeddings.

Classification: enhancement, severity low, priority low.
